### PR TITLE
Cleanup/Partial revert of RGB blood ""rework""

### PR DESCRIPTION
**Pull request by silicons**
## About The Pull Request

I'm going to not shitpost in commits for once. They will contain all the information of what's happening.
Notable things:
blood decals/etc don't have max bloodiness by default anymore (holy shit why)
gibs no longer have max blood by default (holy shit why)
bunch of bugs with forensics and lists fixed
slgiht optimizations and tossed out some unused things
reverts an undocumented but probably unintentional voodoo doll buff
fixes infinite gib streaking

EDIT: Instead, rebuffed iron to +0.25 from 0.01.
Overdose stays.

## Why It's Good For The Game

Too many undocumented/underdocumented mechanical changes from a PR that is described to be mostly cosmetics, too many months of tears due to balance issues from said changes after the person who did them left, too many issues. Gib streaking, gibs having massive amounts of free blood for cults, unoptimized code, so hey here I am going through this line by line. 

## Changelog
Pending
:cl:
refactor: A good amount of the blood RGB rework was cleaned up/reverted, with some notable gameplay changes including: Gibs and blood not having max blood by default (no more easy rampages, cultists), infinite gib streaking, etc etc. 
/:cl:

### CODE REVIEW

**Comment by IronEleven** [comment]
For the record, I think simplemobs bleeding was something Artur did for bloodsuckers.

File: modular_citadel/code/modules/reagents/reagents/cit_reagents.dm [hunk 1]
```diff
@@ -1,7 +1,7 @@
 //body bluids
 /datum/reagent/consumable/semen
 	name = "Semen"
-	description = "Sperm from some animal. I bet you'll drink this out of a bucket someday."
```
Comment (silicons) [inline modular_citadel/code/modules/reagents/reagents/cit_reagents.dm:4]: oh yeah that was changed in the pr too and god this is untasteful
am i just anti fun? :+1: or :-1:

Comment (Trilbyspaceclone) [reply modular_citadel/code/modules/reagents/reagents/cit_reagents.dm:4]: https://github.com/Citadel-Station-13/Citadel-Station-13/pull/5828/files#diff-abb1031157e2ebd499dfd384b4d8da9eR5
Been Oldest commit shows its been their for ages
"In the pr too" - that is wrong

**Comment by silicons** [comment]
Ah yes, simplemobs are the bloodsuckers thing.
Now to find the gib issues.

File: code/modules/mob/living/carbon/damage_procs.dm [hunk 2]
```diff
@@ -87,7 +87,7 @@
 	if(!forced && HAS_TRAIT(src, TRAIT_TOXINLOVER)) //damage becomes healing and healing becomes damage
 		amount = -amount
 		if(amount > 0)
-			blood_volume -= 3*amount // x5 is too much, x3 should be still penalizing enough.
```
Comment (silicons) [inline code/modules/mob/living/carbon/damage_procs.dm:90]: This is the only one that I'm hesitant on, it's only in here because the point of this is to yank out every undocumented change.

Need comments on if this should be kept.

File: modular_citadel/code/modules/reagents/reagents/cit_reagents.dm (hunk 1 above)
Comment (Linzolle) [reply modular_citadel/code/modules/reagents/reagents/cit_reagents.dm:4]: i did not like the comment in the description here anyway

**Comment by Poojawa** [comment]
woulda been nice to have had this level of review when it was still in the works. /shrug

Also it wasn't me that claimed it was just cosmetics, that was you and Bhjin. It was always a rework due to new blood types and new ways of handling them.

File: code/game/objects/items/devices/scanners.dm [hunk 3]
```diff
@@ -407,19 +407,19 @@ SLIME SCANNER
...
+		var/blood_typepath = C.get_blood_id()
+		if(blood_typepath)
 			if(ishuman(C))
 				if(H.bleed_rate)
 					msg += "<span class='danger'>Subject is bleeding!</span>\n"
 			var/blood_percent =  round((C.scan_blood_volume() / (BLOOD_VOLUME_NORMAL * C.blood_ratio))*100)
 			var/blood_type = C.dna.blood_type
-			if(blood_id != ("blood" || "jellyblood"))//special blood substance
-				var/datum/reagent/R = GLOB.chemical_reagents_list[blood_id]
+			if(blood_typepath != /datum/reagent/blood && blood_typepath != /datum/reagent/blood/jellyblood)
```
Comment (Poojawa) [inline code/game/objects/items/devices/scanners.dm]: this probably needs to keep being an Or statement, because not every sample will have both types.

File: code/modules/mob/living/carbon/damage_procs.dm (hunk 2 above)
Comment (Poojawa) [reply code/modules/mob/living/carbon/damage_procs.dm:90]: This was a change I had talked about in public about, more as a mercy.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 4]
```diff
@@ -1051,14 +1051,13 @@
...
 	taste_description = "iron"
 	pH = 6
-	overdose_threshold = 30
 	color = "#c2391d"
 
 /datum/reagent/iron/on_mob_life(mob/living/carbon/C)
 	if((HAS_TRAIT(C, TRAIT_NOMARROW)))
 		return
 	if(C.blood_volume < (BLOOD_VOLUME_NORMAL*C.blood_ratio))
-		C.blood_volume += 0.01 //we'll have synthetics from medbay.
```
Comment (Poojawa) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:1061]: As commented, and discussed in public. hardly an 'undocumented change'.

File: code/modules/surgery/organs/heart.dm [hunk 5]
```diff
@@ -220,21 +220,12 @@ obj/item/organ/heart/cybernetic/upgraded/on_life()
...
 
-
-
 /obj/item/organ/heart/ipc
 	name = "IPC heart"
 	desc = "An electronic pump that regulates hydraulic functions, they have an auto-restart after EMPs."
 	icon_state = "heart-c"
 	organ_flags = ORGAN_SYNTHETIC
 
-/obj/item/organ/heart/ipc/emp_act()
```
Comment (Poojawa) [inline code/modules/surgery/organs/heart.dm:231]: This was discussed. It allows the IPC hearts to restart when ion'd. Why are you removing this?

**Review by Poojawa — CHANGES_REQUESTED** [review]
I notice literally none of this affects the actual reason for your salt, that being Botany blood tomatoes. which should never have actually given usable blood at all and always produce completely full layers of gibs. Normal people gibs do not do this.

Blood Tomatoes were untouched by my PR because of out of scope and the likely shit storm of 'OH BOY POOJ IS NERFING SHIT AGAIN'. Didn't want to deal with it, do the work, and get it ignored afterwards anyway. other people were encouraged to. 

Iron was made dangerous because otherwise taking 300u of pills being able to just ignore your blood levels as any species completely defeats the reasoning behind having specific blood types and making replacement its own thing. It was also discussed at length.

File: code/modules/surgery/organs/heart.dm (hunk 5 above)
Comment (silicons) [reply code/modules/surgery/organs/heart.dm:231]: Because this was not mentioned in the PR title, body, or any comments.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 4 above)
Comment (silicons) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm:1061]: Someone going through files changed and pointing it out on the PR is not the kind of criteria I'd want honestly but sure, my fault for not looking closer.

File: code/game/objects/items/devices/scanners.dm (hunk 3 above)
Comment (silicons) [reply code/game/objects/items/devices/scanners.dm]: The previous statement checked if the blood ID was neither blood nor jellyblood, the statement I replaced it with does the same, I don't even think the previous statement worked as ("blood" || "jellyblood") evaluaes to "blood" because of how the or operator actually works.

**Comment by silicons** [comment]
> I notice literally none of this affects the actual reason for your salt, that being Botany blood tomatoes. which should never have actually given usable blood at all and always produce completely full layers of gibs. Normal people gibs do not do this.

image

They do, you set ALL blood decals bloodiness to max bloodiness and this is causing a massive headache with cult and other things.

**Comment by silicons** [comment]
Regardless, I will reset the iron to some other values in a bit and the *3 can honestly stay.
IPC hearts I'm more against because IPCs have never had EMP vulnerability or anything. Why would they have this of all things from a mostly cosmetic rework?

**Comment by Linzolle** [comment]
I'm with Kevin on the IPCs, they shouldn't be subject to this one thing from EMPs if it doesn't affect anything else with them.

**Comment by silicons** [comment]
exotic blood loss from tox heal is now 3x again down from 5x
iron overdose is back with iron giving +0.25 blood instead of 0.5

**Comment by Poojawa** [comment]
also slightly off topic but sorry for suddenly coming out swinging. just the tone that's been in these PRs was the little thing that pushed my stress over.

basically dealing with the results of failed chemo treatments, no transplant, and what is now her at home with EoL hospice care. It's been a rough weekend so far.

**Comment by Poojawa** [comment]
> 
> 
> I'm with Kevin on the IPCs, they shouldn't be subject to this one thing from EMPs if it doesn't affect anything else with them.

I thought it would make for an interesting quirk they needed to deal with, since I had plans to make them less human in general.

**Comment by silicons** [comment]
Yeah I concede I could be less terse, sorry as well. It's been a rough few months and there's been issues everywhere regarding citcode. I hope things go well with whatever's going on fam.

**Comment by Poojawa** [comment]
Understandable, I know my shitcode is pretty ham-fisted and frustrating to deal with. 

Sadly, as of this post and ~6 hours ago it all came to an end. But it was peaceful and in her sleep.

File: code/game/objects/items/devices/scanners.dm [hunk 6]
```diff
@@ -407,19 +407,19 @@ SLIME SCANNER
...
+				var/datum/reagent/R = GLOB.chemical_reagents_list[blood_typepath]
 				if(R)
 					blood_type = R.name
 				else
-					blood_type = blood_id
+					blood_type = blood_typepath
```
Comment (Ghommie) [inline code/game/objects/items/devices/scanners.dm]: Just replace lines 417-422 with:
```
				var/datum/reagent/R = GLOB.chemical_reagents_list[blood_typepath]
				if(R)
					blood_type = R.name
```
That was something I forgot to change on the ported reagent ID refactor.

File: code/modules/mob/living/blood.dm [hunk 7]
```diff
@@ -29,13 +29,10 @@
 // Takes care blood loss and regeneration
 /mob/living/carbon/human/handle_blood()
 
-	if(NOBLOOD in dna.species.species_traits)
+	if((NOBLOOD in dna.species.species_traits) || (bleed_rate < 0))
```
Comment (Ghommie) [inline code/modules/mob/living/blood.dm]: This will cause the handle_blood cycle to be skipped on negative bleed rate. Undo this change.

File: code/modules/mob/living/blood.dm [hunk 8]
```diff
@@ -29,13 +29,10 @@
...
 		bleed_rate = 0
 		return
 
-	if(bleed_rate <= 0)
```
Comment (Ghommie) [inline code/modules/mob/living/blood.dm:36]: But yes, this should be `if(bleed_rate < 0)`

File: modular_citadel/code/modules/reagents/reagents/cit_reagents.dm [hunk 9]
```diff
@@ -65,15 +65,15 @@
 	blood_state = null
 	bloodiness = null
 
-/obj/effect/decal/cleanable/femcum/New()
-	..()
-	dir = pick(1,2,4,8)
+/obj/effect/decal/cleanable/semen/Initialize(mapload)
```
Comment (Ghommie) [inline modular_citadel/code/modules/reagents/reagents/cit_reagents.dm]:
```suggestion
/obj/effect/decal/cleanable/femcum/Initialize(mapload)
```

**Review by Ghommie — CHANGES_REQUESTED** [review]
Nice.

File: modular_citadel/code/modules/reagents/reagents/cit_reagents.dm (hunk 1 above)
Comment (Ghommie) [reply modular_citadel/code/modules/reagents/reagents/cit_reagents.dm:4]: Whatever, the current comment is too vulgarly smut, good revert.

**Review by Ghommie — APPROVED** [review]
